be/src/agent/heartbeat_server.cpp

```diff
@@ -132,6 +141,12 @@ Status HeartbeatServer::_heartbeat(
         }
     }
 
+    if (need_report) {
+        LOG(INFO) << "Master FE is changed or restarted. report tablet and disk info immediately";
+        std::unique_lock<std::mutex> lk(OLAPEngine::get_instance()->report_mtx);
+        OLAPEngine::get_instance()->report_cv.notify_all();
+    }
```

Comment:
1. `notify_all` needn't to hold mutex first.
2. `report_cv.notify_all()` should encapsulate in a function
3. you can use `_olap_engine` instead of `OLAPEngine::get_instance()`

Reply:
fixed